### PR TITLE
Update dependencies

### PR DESCRIPTION
**Pull request by mchv**
## What does this change?

_Similar to https://github.com/guardian/flexible-feeds/pull/76_

- Upgrade `scala` version to lastest `2.13.x` — a2a75a222ec48ef595b3e2ab75e75af7c495c56b
- Upgrade `sbt` version to lastest `1.10.6` — a6bf349ff021d1041731ba7301155ceb1916fd5c
- Upgrade to Play `3.0.6` — b8195b47066eb551f695485b31ae303769db17c3
- Upgrade to Java `21` using [amigo recipe](https://amigo.gutools.co.uk/recipes/editorial-tools-focal-java21) — 9cb31e0fba18473d5cfa4a3cbfe2e115864b3632
- Upgrade github actions to latest — 19410cfe94bfe0eeaeb83450655e9321a864ee44

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -38,12 +38,12 @@ lazy val pathManager = project.in(file("path-manager"))
       "-J-XX:+PrintGCDetails",
```
Comment (Fweddi) [inline build.sbt:37]: The GC option `-J-XX:+UseConcMarkSweepGC` has been deprecated and causes a fatal exception in recent Java versions.

Removing the option allows us to deploy successfully ([Commit](https://github.com/guardian/path-manager/pull/70/commits/55af041e81e60f9b290ddc941764d9c845bab5d0), [RiffRaff](https://riffraff.gutools.co.uk/deployment/view/4a5e4391-5e4a-4c17-abf6-55b680444919), [Logs](https://logs.gutools.co.uk/s/editorial-tools/app/r/s/etOCU)).

As I understand it, GC now defaults to G1, which should be fine - some comparison of the different GCs can be found [here](https://kstefanj.github.io/2023/12/13/jdk-21-the-gcs-keep-getting-better.html).

**Review by Fweddi — CHANGES_REQUESTED** [review]

File: build.sbt (hunk 1 above)
Comment (mchv) [reply build.sbt:37]: Great point, let me amend and remove this!

Comment (mchv) [reply build.sbt:37]: @Fweddi I have also replaced outdated `MaxRAMFraction` and `InitialRAMFraction ` options (See [details](https://developers.redhat.com/articles/2022/04/19/java-17-whats-new-openjdks-container-awareness#))

**Review by Fweddi — APPROVED** [review]
Deployed successfully, LGTM 👍

**Pull request merged**